libutil/dr_config.c

```diff
@@ -1154,18 +1167,43 @@ dr_register_process(const char *process_name, process_id_t pid, bool global,
     if (status != DR_SUCCESS)
         return status;
 
-    /* set the autoinject string (i.e., path to dynamorio.dll */
-    if (debug) {
-        if (!platform_is_64bit(dr_platform))
-            _sntprintf(wbuf, MAXIMUM_PATH, _TEXT(TSTR_FMT) DEBUG32_DLL, dr_root_dir);
-        else
-            _sntprintf(wbuf, MAXIMUM_PATH, _TEXT(TSTR_FMT) DEBUG64_DLL, dr_root_dir);
+        /* set the autoinject string (i.e., path to dynamorio.dll */
+        /* TODO: cleanup and simplify */
```

Comment:
This is a very fragile piece of code and very hard to test for all corner cases (wstrings, utf8, etc...). That's why I just copied it. It would be better completely rewrite this in a way that it reads the path from a configuration file.

My initial idea to make everything runtime-configurable by adding the `-dr_root=/` is only a partial solution as I learned that install paths end up hard-coded in the binary. This should be fixed independently of this PR, but probably is a pre condition to get this merged eventually. The current solutions are too hacky.